components/service/glean/src/main/java/mozilla/components/service/glean/net/HttpPingUploader.kt

```diff
@@ -36,71 +37,82 @@ class HttpPingUploader(configuration: Configuration) : PingUploader {
      */
     @Suppress("ReturnCount", "MagicNumber")
     override fun upload(path: String, data: String): Boolean {
-        var connection: HttpURLConnection? = null
-        try {
-            connection = openConnection(config.serverEndpoint, path)
-            connection.requestMethod = "POST"
-            connection.connectTimeout = config.connectionTimeout
-            connection.readTimeout = config.readTimeout
-            connection.doOutput = true
-
-            connection.setRequestProperty("Content-Type", "application/json; charset=utf-8")
-            connection.setRequestProperty("User-Agent", config.userAgent)
-            connection.setRequestProperty("Date", createDateHeaderValue())
-
-            // Add headers for supporting the legacy pipeline.
-            connection.setRequestProperty("X-Client-Type", "Glean")
-            connection.setRequestProperty("X-Client-Version", BuildConfig.LIBRARY_VERSION)
-
-            val responseCode = doUpload(connection, data)
-
-            logger.debug("Ping upload: $responseCode")
-
-            when (responseCode) {
-                in HttpURLConnection.HTTP_OK..(HttpURLConnection.HTTP_OK + 99) -> {
-                    // Known success errors (2xx):
-                    // 200 - OK. Request accepted into the pipeline.
-
-                    // We treat all success codes as successful upload even though we only expect 200.
-                    logger.debug("Ping successfully sent ($responseCode)")
-                    return true
-                }
-                in HttpURLConnection.HTTP_BAD_REQUEST..(HttpURLConnection.HTTP_BAD_REQUEST + 99) -> {
-                    // Known client (4xx) errors:
-                    // 404 - not found - POST/PUT to an unknown namespace
-                    // 405 - wrong request type (anything other than POST/PUT)
-                    // 411 - missing content-length header
-                    // 413 - request body too large (Note that if we have badly-behaved clients that
-                    //       retry on 4XX, we should send back 202 on body/path too long).
-                    // 414 - request path too long (See above)
-
-                    // Something our client did is not correct. It's unlikely that the client is going
-                    // to recover from this by re-trying again, so we just log and error and report a
-                    // successful upload to the service.
-                    logger.error("Server returned client error code: $responseCode")
-                    return true
-                }
-                else -> {
-                    // Known other errors:
-                    // 500 - internal error
+        if (config.logPings) {
+            // Parse and reserialize the JSON so it has indentation and is human-readable.
+            var json = JSONObject(data)
```

Comment:
This and the next line can throw. I'll wrap them in a try catch and watch out for JSONException.